### PR TITLE
HDDS-1974. Implement OM CancelDelegationToken request to use Cache and DoubleBuffer.

### PR DESCRIPTION
**Pull request by bharatviswa504**

### CODE REVIEW

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/OzoneDelegationTokenSecretManager.java [hunk 1]
```diff
@@ -287,18 +287,40 @@ public OzoneTokenIdentifier cancelToken(Token<OzoneTokenIdentifier> token,
...
+        LOG.error("Unable to remove token " + id.getSequenceNumber(), e);
+      }
+      TokenInfo info = currentTokens.remove(id);
+      if (info == null) {
+        throw new InvalidToken("Token not found " + formatTokenId(id));
+      }
+    } else {
+      // Check whether token is there in-memory map of tokens or not on the
+      // OM leader.
+      TokenInfo info = currentTokens.get(id);
```
Comment (arp7) [inline hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/OzoneDelegationTokenSecretManager.java]: Is the in-memory map a complete map?

Comment (bharatviswa504) [reply hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/OzoneDelegationTokenSecretManager.java]: Yes, a complete map.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 199 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 44 | Maven dependency ordering for branch |
| -1 | mvninstall | 34 | hadoop-ozone in trunk failed. |
| -1 | compile | 38 | hadoop-hdds in trunk failed. |
| -1 | compile | 33 | hadoop-ozone in trunk failed. |
| -0 | checkstyle | 27 | The patch fails to run checkstyle in hadoop-hdds |
| -0 | checkstyle | 24 | The patch fails to run checkstyle in hadoop-ozone |
| +1 | mvnsite | 0 | trunk passed |
| -1 | shadedclient | 136 | branch has errors when building and testing our client artifacts. |
| -1 | javadoc | 36 | hadoop-hdds in trunk failed. |
| 0 | spotbugs | 1110 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 90 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 658 | hadoop-ozone in trunk failed. |
| -0 | patch | 1210 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 48 | Maven dependency ordering for patch |
| -1 | mvninstall | 39 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 49 | hadoop-ozone in the patch failed. |
| -1 | compile | 161 | hadoop-hdds in the patch failed. |
| -1 | compile | 46 | hadoop-ozone in the patch failed. |
| -1 | javac | 161 | hadoop-hdds in the patch failed. |
| -1 | javac | 46 | hadoop-ozone in the patch failed. |
| -0 | checkstyle | 44 | The patch fails to run checkstyle in hadoop-ozone |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 1 | The patch has no whitespace issues. |
| +1 | shadedclient | 50 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 129 | hadoop-hdds generated 16 new + 0 unchanged - 0 fixed = 16 total (was 0) |
| +1 | findbugs | 1047 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 495 | hadoop-hdds in the patch passed. |
| -1 | unit | 1554 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 74 | The patch does not generate ASF License warnings. |
| | | 5996 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.om.TestOMRatisSnapshots |
|   | hadoop.ozone.om.TestOzoneManagerRestart |
|   | hadoop.ozone.om.TestOzoneManagerRestInterface |
|   | hadoop.ozone.om.TestOzoneManager |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1308 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 55283ee71c29 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 8d754c2 |
| Default Java | 1.8.0_212 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/branch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out//home/jenkins/jenkins-slave/workspace/hadoop-multibranch_PR-1308/out/maven-branch-checkstyle-hadoop-hdds.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out//home/jenkins/jenkins-slave/workspace/hadoop-multibranch_PR-1308/out/maven-branch-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/branch-javadoc-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/patch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out//home/jenkins/jenkins-slave/workspace/hadoop-multibranch_PR-1308/out/maven-patch-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/diff-javadoc-javadoc-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/testReport/ |
| Max. process+thread count | 3555 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/3/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 105 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 31 | Maven dependency ordering for branch |
| +1 | mvninstall | 606 | trunk passed |
| +1 | compile | 367 | trunk passed |
| +1 | checkstyle | 66 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 911 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 170 | trunk passed |
| 0 | spotbugs | 469 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 689 | trunk passed |
| -0 | patch | 508 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 19 | Maven dependency ordering for patch |
| +1 | mvninstall | 592 | the patch passed |
| +1 | compile | 428 | the patch passed |
| +1 | javac | 428 | the patch passed |
| -0 | checkstyle | 46 | hadoop-ozone: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 815 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 189 | the patch passed |
| -1 | findbugs | 466 | hadoop-ozone generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0) |
||| _ Other Tests _ |
| +1 | unit | 349 | hadoop-hdds in the patch passed. |
| -1 | unit | 2203 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 43 | The patch does not generate ASF License warnings. |
| | | 8580 | |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:hadoop-ozone |
|  |  Null passed for non-null parameter of new org.apache.hadoop.ozone.om.response.security.OMCancelDelegationTokenResponse(OzoneTokenIdentifier, OzoneManagerProtocolProtos$OMResponse) in org.apache.hadoop.ozone.om.request.security.OMCancelDelegationTokenRequest.validateAndUpdateCache(OzoneManager, long, OzoneManagerDoubleBufferHelper)  At OMCancelDelegationTokenRequest.java:of new org.apache.hadoop.ozone.om.response.security.OMCancelDelegationTokenResponse(OzoneTokenIdentifier, OzoneManagerProtocolProtos$OMResponse) in org.apache.hadoop.ozone.om.request.security.OMCancelDelegationTokenRequest.validateAndUpdateCache(OzoneManager, long, OzoneManagerDoubleBufferHelper)  At OMCancelDelegationTokenRequest.java:[line 101] |
|  |  Null passed for non-null parameter of new org.apache.hadoop.ozone.om.response.security.OMGetDelegationTokenResponse(OzoneTokenIdentifier, long, OzoneManagerProtocolProtos$OMResponse) in org.apache.hadoop.ozone.om.request.security.OMGetDelegationTokenRequest.validateAndUpdateCache(OzoneManager, long, OzoneManagerDoubleBufferHelper)  At OMGetDelegationTokenRequest.java:of new org.apache.hadoop.ozone.om.response.security.OMGetDelegationTokenResponse(OzoneTokenIdentifier, long, OzoneManagerProtocolProtos$OMResponse) in org.apache.hadoop.ozone.om.request.security.OMGetDelegationTokenRequest.validateAndUpdateCache(OzoneManager, long, OzoneManagerDoubleBufferHelper)  At OMGetDelegationTokenRequest.java:[line 140] |
| Failed junit tests | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientForAclAuditLog |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1308 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 9186e5941d4c 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 8d754c2 |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/1/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/1/artifact/out/new-findbugs-hadoop-ozone.html |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/1/testReport/ |
| Max. process+thread count | 5368 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 39 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for branch |
| +1 | mvninstall | 583 | trunk passed |
| +1 | compile | 342 | trunk passed |
| +1 | checkstyle | 61 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 808 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 168 | trunk passed |
| 0 | spotbugs | 457 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 674 | trunk passed |
| -0 | patch | 490 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 16 | Maven dependency ordering for patch |
| +1 | mvninstall | 615 | the patch passed |
| +1 | compile | 382 | the patch passed |
| +1 | javac | 382 | the patch passed |
| -0 | checkstyle | 38 | hadoop-ozone: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 667 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 165 | the patch passed |
| -1 | findbugs | 471 | hadoop-ozone generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0) |
||| _ Other Tests _ |
| +1 | unit | 293 | hadoop-hdds in the patch passed. |
| -1 | unit | 2149 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 40 | The patch does not generate ASF License warnings. |
| | | 7954 | |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:hadoop-ozone |
|  |  Null passed for non-null parameter of new org.apache.hadoop.ozone.om.response.security.OMCancelDelegationTokenResponse(OzoneTokenIdentifier, OzoneManagerProtocolProtos$OMResponse) in org.apache.hadoop.ozone.om.request.security.OMCancelDelegationTokenRequest.validateAndUpdateCache(OzoneManager, long, OzoneManagerDoubleBufferHelper)  At OMCancelDelegationTokenRequest.java:of new org.apache.hadoop.ozone.om.response.security.OMCancelDelegationTokenResponse(OzoneTokenIdentifier, OzoneManagerProtocolProtos$OMResponse) in org.apache.hadoop.ozone.om.request.security.OMCancelDelegationTokenRequest.validateAndUpdateCache(OzoneManager, long, OzoneManagerDoubleBufferHelper)  At OMCancelDelegationTokenRequest.java:[line 101] |
|  |  Null passed for non-null parameter of new org.apache.hadoop.ozone.om.response.security.OMGetDelegationTokenResponse(OzoneTokenIdentifier, long, OzoneManagerProtocolProtos$OMResponse) in org.apache.hadoop.ozone.om.request.security.OMGetDelegationTokenRequest.validateAndUpdateCache(OzoneManager, long, OzoneManagerDoubleBufferHelper)  At OMGetDelegationTokenRequest.java:of new org.apache.hadoop.ozone.om.response.security.OMGetDelegationTokenResponse(OzoneTokenIdentifier, long, OzoneManagerProtocolProtos$OMResponse) in org.apache.hadoop.ozone.om.request.security.OMGetDelegationTokenRequest.validateAndUpdateCache(OzoneManager, long, OzoneManagerDoubleBufferHelper)  At OMGetDelegationTokenRequest.java:[line 140] |
| Failed junit tests | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.TestStorageContainerManager |
|   | hadoop.ozone.TestMiniChaosOzoneCluster |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestFailureHandlingByClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1308 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 0e80f25b8f49 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 8d754c2 |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/4/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/4/artifact/out/new-findbugs-hadoop-ozone.html |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/4/testReport/ |
| Max. process+thread count | 4079 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 85 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 32 | Maven dependency ordering for branch |
| +1 | mvninstall | 649 | trunk passed |
| +1 | compile | 392 | trunk passed |
| +1 | checkstyle | 74 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 965 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 174 | trunk passed |
| 0 | spotbugs | 546 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 797 | trunk passed |
| -0 | patch | 596 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for patch |
| +1 | mvninstall | 696 | the patch passed |
| +1 | compile | 472 | the patch passed |
| +1 | javac | 472 | the patch passed |
| -0 | checkstyle | 52 | hadoop-ozone: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 914 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 205 | the patch passed |
| -1 | findbugs | 597 | hadoop-ozone generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0) |
||| _ Other Tests _ |
| +1 | unit | 413 | hadoop-hdds in the patch passed. |
| -1 | unit | 2493 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 45 | The patch does not generate ASF License warnings. |
| | | 9599 | |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:hadoop-ozone |
|  |  Null passed for non-null parameter of new org.apache.hadoop.ozone.om.response.security.OMCancelDelegationTokenResponse(OzoneTokenIdentifier, OzoneManagerProtocolProtos$OMResponse) in org.apache.hadoop.ozone.om.request.security.OMCancelDelegationTokenRequest.validateAndUpdateCache(OzoneManager, long, OzoneManagerDoubleBufferHelper)  At OMCancelDelegationTokenRequest.java:of new org.apache.hadoop.ozone.om.response.security.OMCancelDelegationTokenResponse(OzoneTokenIdentifier, OzoneManagerProtocolProtos$OMResponse) in org.apache.hadoop.ozone.om.request.security.OMCancelDelegationTokenRequest.validateAndUpdateCache(OzoneManager, long, OzoneManagerDoubleBufferHelper)  At OMCancelDelegationTokenRequest.java:[line 101] |
|  |  Null passed for non-null parameter of new org.apache.hadoop.ozone.om.response.security.OMGetDelegationTokenResponse(OzoneTokenIdentifier, long, OzoneManagerProtocolProtos$OMResponse) in org.apache.hadoop.ozone.om.request.security.OMGetDelegationTokenRequest.validateAndUpdateCache(OzoneManager, long, OzoneManagerDoubleBufferHelper)  At OMGetDelegationTokenRequest.java:of new org.apache.hadoop.ozone.om.response.security.OMGetDelegationTokenResponse(OzoneTokenIdentifier, long, OzoneManagerProtocolProtos$OMResponse) in org.apache.hadoop.ozone.om.request.security.OMGetDelegationTokenRequest.validateAndUpdateCache(OzoneManager, long, OzoneManagerDoubleBufferHelper)  At OMGetDelegationTokenRequest.java:[line 140] |
| Failed junit tests | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.om.TestSecureOzoneManager |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineProvider |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.om.TestOzoneManagerHA |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1308 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 410728f6f9e3 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 8d754c2 |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/2/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/2/artifact/out/new-findbugs-hadoop-ozone.html |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/2/testReport/ |
| Max. process+thread count | 5310 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 142 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 33 | Maven dependency ordering for branch |
| +1 | mvninstall | 745 | trunk passed |
| +1 | compile | 418 | trunk passed |
| +1 | checkstyle | 93 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1078 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 178 | trunk passed |
| 0 | spotbugs | 454 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 670 | trunk passed |
| -0 | patch | 494 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 21 | Maven dependency ordering for patch |
| +1 | mvninstall | 562 | the patch passed |
| +1 | compile | 373 | the patch passed |
| +1 | javac | 373 | the patch passed |
| +1 | checkstyle | 75 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 747 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 166 | the patch passed |
| +1 | findbugs | 657 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 342 | hadoop-hdds in the patch passed. |
| -1 | unit | 2250 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 45 | The patch does not generate ASF License warnings. |
| | | 8733 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientForAclAuditLog |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1308 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux c2d079d529d4 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / d873ddd |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/5/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/5/testReport/ |
| Max. process+thread count | 5034 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1308/5/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by bharatviswa504** [comment]
Test failures are not related to this patch.
Thank You @arp7 for the review.
I will commit this to the trunk.

**Pull request merged**